### DISCUSSION TITLE
Best practice for building with extensions?

### DISCUSSION THREAD

**Discussion by marklit** [duckdb/duckdb-spatial#106] [Q&A]
When I want to re-build from the main branch I'm often presented with the following when I then launch the new build.

```sql
INSTALL inet;
```

```
Error: HTTP Error: Failed to download extension "inet" at URL "http://extensions.duckdb.org/cc60796b7a/linux_amd64/inet.duckdb_extension.gz"
Extension "inet" is an existing extension.

Are you using a development build? In this case, extensions might not (yet) be uploaded.
```

I want to offer my readers an alternative where they can build extensions used in posts alongside the rest of DuckDB and the Spatial extension. Is the following a good one-line change to the build instructions for doing so or is there a better way?

```bash
$ CLIENT_FLAGS="-DBUILD_EXTENSIONS='parquet;json;inet'" GEN=ninja make release
```

**Comment by Maxxen** [comment]
Probably, we're thinking of ways to allow people to "mix and match" the DuckDB features(extensions) they want from the main repo, but for now modifying the makefile is the way to go.
